variables.md

```diff
@@ -0,0 +1,204 @@
+
+
+#### Disimilarities between ```var``` and ```let``` keywords
+
+the difference between them is that ```var``` is function scoped and ```let``` is block scoped. example:
+
+```
+console.log(x);
+var x=5;
+console.log(x); //Output: undefined, 5
+```
+
+if an undeclared variable is logged to the console and later declared with ```var``` and assigned a value, it will return undefine and the value assigned to it.
+
+an error will be returned if let is used in declaring such variable. example:
+
+
+```
+console.log(x); // output: ReferenceError: x is not defined
+let x=5;
+console.log(x); //Output: 5
+```
+
+### ```const```
+
+Just like with ```var``` and ```let``` you can store any value in a ```const``` variable. The way you declare a const variable and assign a value to it follows the same structure as ```let``` and ```var```.
+
+```
+const myName = 'Gilberto';
+console.log(myName); // Output: Gilberto
+```
+
+However, a const variable cannot be reassigned because it is constant. If you try to reassign a const variable, you’ll get a TypeError.
+
+Constant variables must be assigned a value when declared. If you try to declare a const variable without a value, you’ll get a SyntaxError.
+
+```
+NOTE: ```let``` is the preferred way to declare a variable when it can be reassigned, and ```const``` is the preferred way to declare a variable with a constant value.
+```
+
+
+### General Rules for Naming Variables:
+
+- Variable names cannot start with numbers.
+
+- Variable names are case sensitive, so myName and myname would be different variables. It is bad practice to create two variables that have the same name using different cases.
+
+- Variable names cannot be the same as keywords. For a comprehensive list of keywords check out [MDN’s keyword documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords).
+
+
+
+## Mathematical Assignment Operators
+
+Variables and math operators can be used to calculate new values and assign them to a variable.
+
+```
+let w = 4;
+w = w + 1;
+
+console.log(w); // Output: 5
+```
+
+In the example above, a variable ```w``` is created with the number ```4``` assigned to it. The following line, ```w = w + 1```, increases the value of ```w``` from ```4``` to ```5```.
+
+Another way that ```w``` could have been reassigned after performing some mathematical operation on it is to use built-in mathematical assignment operators. We could re-write the code above to be:
+
+```
+let w = 4;
+w += 1;
+
+console.log(w); // Output: 5
+```
+
+In the second example, the ```+=``` assignment operator is used to reassign ```w```. We’re performing the mathematical operation of the first operator ```+``` using the number to the right, then reassigning ```w``` to the computed value.
+
+We also have access to other mathematical assignment operators: ```-=```, ```*=```, and ```/=``` which work in a similar fashion.
+
+```
```

Comment:
Please remember about the `javascript` keyword for your code blocks
Do this for all other occurrences

```suggestion
```javascript
```